xlators/storage/posix/src/posix-helpers.c

```diff
@@ -1693,7 +1693,7 @@ is_fresh_file(struct timespec *ts)
     int64_t elapsed;
 
     timespec_now_realtime(&now);
-    elapsed = (int64_t)gf_tsdiff(ts, &now);
+    elapsed = gf_tsdiff(ts, &now);
```

Comment:
I've just seen that we have an issue here from commit 2c349f93. The elapsed time is expected to be represented in microseconds, but that patch changed it to nanoseconds without changing the value in line 1706. We should compare `elapsed` with 1000000000 instead of 1000000.

Can you also update this in this patch ? otherwise it's also fine to merge this patch and fix the problem in another patch.

Reply:
Sure!! I'll change it now.

Reply:
Thanks a lot.

Reply:
No problem!